### ISSUE TITLE
houdini to katana asset published 

### ISSUE THREAD

**Issue by ronnyascencio** [masqu3rad3/tik_manager4#100]
**Describe the bug**
the name and attributes should match from one dcc to an other 

**To Reproduce**
1. Go to 'Publish'

**Expected behavior**
if i publish an asset from houdini this should have the attrebutes and name embed  for now the published asset has a tik manager name and is merged and has not any of the attributes builded inside houdini  when i transfer the asset or impor to Katana 

**Screenshots**
![Screenshot from 2024-04-23 08-51-19](https://github.com/masqu3rad3/tik_manager4/assets/95187946/c58b8230-b8ae-44dd-8d24-0daa9a371e02)
![Screenshot from 2024-04-23 08-51-06](https://github.com/masqu3rad3/tik_manager4/assets/95187946/01a13583-9410-4594-8c01-bed1e20b1caa)

**DCC**
Please complete the following information:
- Dcc name(s): Houdini / katana
- Dcc Version(s) houdini 19.5.805  katana 6.5v4

**Desktop (please complete the following information):**
 - OS: linux
 - Version rocky 8.9

**Comment by masqu3rad3** [comment]
Apologies, I couldn't understand the issue in here.

Which formats are we talking about? Tik Manager only renames the nodes that hold the files in it (importamatic node in Katana and File node in Houdini) It is not changing the path attributes so we cannot treat it as a USD authoring tool. USD authoring is fairly custom for each studio and can be supported in Tik Manager with custom extractors and ingestors.

Can you elaborate the request a little bit more?

**Comment by ronnyascencio** [comment]
1 the rename of the importomatic node in katana is not the problem  

in this or any other case when you have an asset you set some attributes like faces selected for aply specific materials   in the same model  

so I have an asset  with this attributes:
material 1
materia 2

this attributes should be transfered when it is exported   and when we use the  Tik publisher  it export an Alembic  but this alembic that is exported  is merged  and overriten all attributs and name set in houdini 

tik_export
![tik_export](https://github.com/masqu3rad3/tik_manager4/assets/95187946/13757d5c-0143-48ce-bb5d-da1e0dafdc8a)

my export

![my_export](https://github.com/masqu3rad3/tik_manager4/assets/95187946/1846c0a5-225c-4b9b-bce0-3143d267ed7c)

**Comment by masqu3rad3** [comment]
I see. This is related with the houdini alembic extractor.
Extractors in tik manager are mostly for generic use.

Can you describe how you export the alembic from the houdini?

Alternatively, did you try the USD instead?

**Comment by ronnyascencio** [comment]
the usd file is kind of working but it generate a prototypes and this shoudlnt be set unless the publish is a instances like for example 100 trees or something like that

**Comment by ronnyascencio** [comment]
Regarding to the export of the alembic and usd  exported from houdini  when is published also is not grabing the animation  so maybe should good to add a check box or something if is an asset animated or not to be export it with that

**Comment by masqu3rad3** [comment]
Extracts are category sensitive.

If you export it from an animation category, they should export the animation.

**Comment by masqu3rad3** [comment]
so essentially, if you export from an Animation, Fx, Layout or Lighting Category, animation should be exported.
For the others, its static.

Again, extracts can be studio/individual specific. Any TD can create a new extractor easily.

**Comment by ronnyascencio** [comment]
yes but i mean my asset is animated it has an  animation an i am  making on houdini some instace of that asset ( animated) then to publishe it i am publishing it under a asset > assambly ( because this is an instace that is what i think ) so how do i need to publish the work  under asset> animation?

**Comment by masqu3rad3** [comment]
ah.. Okay I got it. You have a valid point. However that is such an extreme case to be fair. The usual way is to apply the animation on shot level. Or add some proceduralism with a rig (or rig like structure) which can be triggered (animated) in the shot.

Having that said, your situation can be a good showcase about how flexible can the system be.
The simplest approach would be just exporting the animated asset from a animation category. Animation category.

By default, the animation category is a 'shot' category. Meaning that it will be available only for global and shot tasks.
You can change this from the Settings.
Go to Settings -> Project -> Category Definitions. 
Find the 'Animation' and change its type from 'shot' to the blank one.
Now this category will also be available in asset context. (if you change it to asset, it won't be available in shot context)

The other ways (which requires a bit coding skills) to do this is to either alter the existing extractors (Not recommended since they will be overridden with Tik Manager updates) or adding new ones.
This way either the animation option can be exposed or you can extract anything anywhere anyway you like.

Since your workflow is a bit out of ordinary, I prefer not to change the core according to that. If we start to put too many options into a pipeline, it will eventually lose its purpose.

**Comment by ronnyascencio** [comment]
git it  i actualy did that  export on shot level,  i notice that from houdini the animation is just exported as an alembic and no usd   just for  what i think is a bit fexible  export usd  ( crowd, fx and instances) it takes less in file that a abc  

thanks for the help!

**Comment by masqu3rad3** [comment]
You can define USD export from Animation category too.
Its just couple of clicks.
Go to Settings -> Project -> Category Definitions.
Click to the '+' button next to the extracts and add Usd.

You can do the same in the Settings -> Common -> Category Definirions (Common)
So it will be permanent for all future projects.

It is not complete and much detailed yet but I would suggest you to take a look at the documentation:
https://tik-manager4.readthedocs.io/en/latest/user_guide.html#category-definitions

**Comment by ronnyascencio** [comment]
is working perfect the only think dont like is how the naming is set from houdini whe publish

**Comment by masqu3rad3** [comment]
If you can give me very detailed step by step instructions how you want to extract abc or usd from houdini I can help you to create your own extractor plugin for Tik Manager4. 

Actually, for houdini, it may be better to make a Tik Manager publish HDA which gets saved with the scene. 
I am a bit rusty with hda though, it's been a while since I last did one.
